oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -119,6 +166,15 @@ impl Runtime {
         let module_name = self.configuration.entry_module.clone();
         let entrypoint = self.configuration.entrypoint.clone();
 
+        if cfg!(feature = "oak_debug") {
+            info!("spawning introspection server on new thread");
+            let runtime = self.clone();
+            let _thread_handle = thread::Builder::new()
+                .name("introspection-server".to_string())
+                .spawn(move || introspect::serve(1909, runtime))
```

Comment:
Make the port a flag at some point? Similar to the metrics one I guess. More generally, I think the metrics stuff is conceptually similar to this, but these two features are both starting web servers in different ways (one from main, one from the runtime, one feature gated, the other one not, one with a dedicated flag, the other one not, etc.).

@daviddrysdale @rbehjati could you come up with a consistent way of doing this? Perhaps worth tracking in a separate issue.

Reply:
Added a TODO to make the port externally configurable.

I thought about re-using the metrics server, but it seemed easier to keep them separate so they can be turned on and off independently (feature gated or some other way), and don't have to worry about a shared path namespace.

Reply:
To clarify: I don't think this should reuse the same server, just that it should create a server in a similar way.

Reply:
I agree that it would be nice to keep these servers consistent, although one is only used during the development and the other is more important for production. So that kind of consistency might not be necessary. On the other hand, moving the metrics server into the Runtime might make #899 easier. 
I wonder what is conceptually the right thing to do. Do we want to update [the description of the Oak Runtime](https://github.com/project-oak/oak/blob/master/docs/concepts.md#oak-runtime) to something like the following?

> The **Oak Runtime** is the core software component of Project Oak; it is responsible for executing Oak Modules, enforcing policies on top of data, and producing remote attestations for clients, _as well as serving debug and monitoring information._"

Reply:
@rbehjati thanks for the suggestion, I think updating the description in that direction makes sense, feel free to create a PR and send it out for review!